api/src/pcapi/core/providers/titelive_book_search.py

```diff
@@ -0,0 +1,240 @@
+        return "pornography-or-violence"
+
+    # Toeic or toefl
+    if constants.TOEIC_TEXT in title or constants.TOEFL_TEXT in title:
+        return "toeic-toefl"
+
+    # --- GTL-based categorization ---
+    gtl_id = get_gtl_id(article)
+    if gtl_id == EMPTY_GTL.code:
+        return "empty-gtl"
+
+    gtl_level_01_code = gtl_id[:2]
+
+    # ouvrage du rayon scolaire
+    if gtl_level_01_code == constants.GTL_LEVEL_01_SCHOOL:
+        return "school"
+
+    # ouvrage du rayon parascolaire,
+    # code de la route (méthode d'apprentissage + codes basiques), code nautique, code aviation, etc...
+    if gtl_level_01_code == constants.GTL_LEVEL_01_EXTRACURRICULAR:
+        return "extracurricular"
+
+    # Petite jeunesse (livres pour le bains, peluches, puzzles, etc...)
+    gtl_level_02_code = gtl_id[2:4]
+    if gtl_level_01_code == constants.GTL_LEVEL_01_YOUNG and gtl_level_02_code in [
+        constants.GTL_LEVEL_02_BEFORE_3,
+        constants.GTL_LEVEL_02_AFTER_3_AND_BEFORE_6,
+    ]:
+        return "small-young"
+
+    # --- Default categorization ---
+    if article.codesupport not in constants.TITELIVE_BOOK_SUPPORTS_BY_CODE:
+        # If we find no known support code, the product is deemed ineligible
+        return "no-known-codesupport"
+
+    if not constants.TITELIVE_BOOK_SUPPORTS_BY_CODE[article.codesupport].get("is_allowed"):
+        return "uneligible-product-subcategory"
+
+    return None
+
+
+def get_book_format(codesupport: str | None) -> str | None:
+    if codesupport is None:
+        return None
+    book_format = constants.TITELIVE_BOOK_SUPPORTS_BY_CODE[codesupport].get("book_format")
+    return book_format.value if book_format else None
+
+
+def build_book_extra_data(article: TiteLiveBookArticle, authors: list) -> offers_models.OfferExtraData:
+    gtl_id = get_gtl_id(article)
+
+    rayon = None
+    csr_id = None
+    if gtl_id != EMPTY_GTL.code:
+        csr_label = get_closest_csr(gtl_id)
+        if csr_label is not None:
+            rayon = csr_label.get("label")
+            csr_id = csr_label.get("csr_id")
+
+    result = offers_models.OfferExtraData(
+        author=",".join(authors) if authors else None,
+        bookFormat=get_book_format(article.codesupport),
+        code_clil=article.code_clil,
+        collection=article.collection,
+        comment=article.commentaire,
+        csr_id=csr_id,
+        date_parution=str(article.dateparution) if article.dateparution else None,
+        dispo_label=article.libelledispo,
+        dispo=article.dispo,
+        distributeur=article.distributeur,
+        ean=article.gencod,
+        editeur=article.editeur,
+        gtl_id=gtl_id,
+        langue=article.langue,
+        langueiso=article.langueiso,
+        nb_pages=article.pages,
+        num_in_collection=article.collection_no,
+        prix_livre=article.prix,
+        rayon=rayon,
+        schoolbook=int(article.scolaire) != 0 if article.scolaire is not None else False,
```

Comment:
[détail / peut-être une question goût] On pourrait aussi écrire ça : 

```suggestion
        schoolbook=bool(int(article.scolaire or 0)),  # value can be None
```

... mais c'est peut-être une question de goût. (Je reconnais ne pas être fan des `... if... else...`. ;) )